shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,240 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_ACCESSIBILITY_BRIDGE_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_ACCESSIBILITY_BRIDGE_H_
+
+#include <unordered_map>
+
+#include "flutter/shell/platform/embedder/embedder.h"
+
+#include "flutter/third_party/accessibility/ax/ax_event_generator.h"
+#include "flutter/third_party/accessibility/ax/ax_tree.h"
+#include "flutter/third_party/accessibility/ax/ax_tree_observer.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_delegate.h"
+
+#include "flutter_accessibility.h"
+
+namespace ui {
+
+//------------------------------------------------------------------------------
+/// Use this class to maintain an accessibility tree. This class consumes
+/// semantics updates from the embedder API and produces an accessibility tree
+/// in the native format.
+///
+/// To use this class, you must provide your own implementation of
+/// FlutterAccessibility and AccessibilityBridgeDelegate.
+class AccessibilityBridge : public AXTreeObserver {
+ public:
+  //------------------------------------------------------------------------------
+  /// Delegate to handle accessibility event and route accessibility action
+  /// back to the Flutter framework.
+  ///
+  /// The accessibility events are generated when accessibility tree changes.
+  /// These events must be sent to the native accessibility system through
+  /// the native API for the system to pick up the changes
+  /// (e.g. NSAccessibilityPostNotification in MacOS).
+  ///
+  /// The accessibility actions are generated by the native accessibility system
+  /// when users interacted with the assistive technologies. Those actions
+  /// needed to be sent to the Flutter framework.
+  class AccessibilityBridgeDelegate {
+   public:
+    virtual ~AccessibilityBridgeDelegate() = default;
+    //------------------------------------------------------------------------------
+    /// @brief      Handle accessibility events generated due to accessibility
+    /// tree
+    ///             changes.
```

Comment:
nit: formatting here and below. Looks like clang-format might have added line breaks wher eyou don't want them.

not a nit: these docs need to be improved. If this really is the one coming from Flutter to platform, then we should give some examples of what kind of events might get sent and handled here. We should talk about what a targetted event is - e.g. what can be a target, what kind of events there are.

Also, please avoid using raw pointers - I'm assuming this probably wants to be a `std::shared_ptr<AccessibilityBridge>`, but a weak or unique is fine too if that's what's appropriate.